text/0185-configuration-file.md

```diff
@@ -0,0 +1,98 @@
+
+## 3. Functional Specification
+
+### 3.1. Supported Formats
+
+| Format    |
+|-----------|
+| `.toml`   |
+
+### 3.2. Priority Order
+
+`Configuration file < Env variables < Command-line options`
+
+It means the options in the configuration file are overwritten by the environment variable set (if they exist) and that can be themself overwritten by the command-line options (if they exist).
+
+### 3.3. Default Configuration File
+
+A default config file is defined as `config.toml` and is retrieved by Meilisearch by default in its current working directory.
+
+The default configuration file details all configurations keys, some of them are commented.
+
+### 3.4. `--config-file-path`/`MEILI_CONFIG_FILE_PATH`
+
+It is possible to override the default configuration file by setting the `MEILI_CONFIG_FILE_PATH` environment variable or by specifying the `--config-file-path` option.
+
+### 3.5. Configuration File Content Definition
+
+The current consistency rule is that all environment variables and launch options are configurable by a dedicated key in the configuration file.
+
+Configuration keys are named following the `snake_case` convention from the name of a CLI option. That is, `--import-dump` must be named `import_dump` within the configuration file.
+
+There is an exception rule. The `config_file_path` key is not accepted in the configuration file and stops Meilisearch from being launched if specified.
+
+See [Instance Options](0119-instance-options.md) for more details.
+
+### 3.6. Indication of the loaded configuration file
+
+When Meilisearch is launched, the loaded configuration file is indicated by the `Config file path` key on the stdout.
+
+```bash
+888b     d888          d8b 888 d8b                                            888
+8888b   d8888          Y8P 888 Y8P                                            888
+88888b.d88888              888                                                888
+888Y88888P888  .d88b.  888 888 888 .d8888b   .d88b.   8888b.  888d888 .d8888b 88888b.
+888 Y888P 888 d8P  Y8b 888 888 888 88K      d8P  Y8b     "88b 888P"  d88P"    888 "88b
+888  Y8P  888 88888888 888 888 888 "Y8888b. 88888888 .d888888 888    888      888  888
+888   "   888 Y8b.     888 888 888      X88 Y8b.     888  888 888    Y88b.    888  888
+888       888  "Y8888  888 888 888  88888P'  "Y8888  "Y888888 888     "Y8888P 888  888
+
+Config file path:       "./config.toml"
+```
+
+If Meilisearch can't find the default `config.toml` or if no other config file is specified with `MEILI_CONFIG_FILE_PATH` env var or `--config-file-path` options, `none` is indicated.
+
+### 3.7. Errors
+
+#### 3.7.1. Defining `config_file_path` in the configuration file
+
+Defining `config_file_path` in the configuration file is not valid and Meilisearch returns an error.
+
+```bash
+`config_file_path` is not supported in config file
```

Comment:
```suggestion
Error: `config_file_path` is not supported in config file
```

Reply:
@curquiza Can we consider replacing "config file" by "configuration file" in that error message to be consistent with the others?